src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    @Test
+    void save_withoutEmail_notOk() {
+        User user = new User();
+        user.setPassword("1234");
+        Assertions.assertThrows(DataProcessingException.class, () -> userDao.save(user),
+                "You can't add a user without an email, please fixed this");
+    }
+
+    @Test
+    void findByEmail_correctEmail_ok() {
+        userDao.save(defaultUser);
+        User actual = userDao.findByEmail(defaultUser.getEmail()).orElseGet(
+                () -> Assertions.fail("User must be find by email: "
+                        + defaultUser.getEmail()));
+        Assertions.assertEquals(defaultUser.getId(), actual.getId(),
+                "User id should be: " + defaultUser.getId());
+        Assertions.assertEquals(defaultUser.getEmail(), actual.getEmail(),
+                "User email should be: " + defaultUser.getEmail());
+    }
+
+    @Test
+    void findByEmail_nullEmail_notOk() {
+        userDao.save(defaultUser);
+        Assertions.assertThrows(NoSuchElementException.class,
+                () -> userDao.findByEmail(null).get(),
+                "There should not be a user with email null");
+    }
+
+    @Test
+    void findByEmail_incorrectEmail_notOk() {
+        userDao.save(defaultUser);
+        Assertions.assertThrows(NoSuchElementException.class,
+                () -> userDao.findByEmail("invalid email").get(),
+                "There should not be a user with invalid email");
+    }
+
+    @Test
+    void findById_correctCase_ok() {
+        User user = new User();
+        user.setEmail("user@gmail.com");
+        user.setPassword("1234");
+        userDao.save(defaultUser);
+        userDao.save(user);
+        Assertions.assertEquals(defaultUser.getEmail(), userDao.findById(
+                defaultUser.getId()).get().getEmail());
+        Assertions.assertEquals(user.getEmail(), userDao.findById(
+                user.getId()).get().getEmail());
+    }
+
+    @Test
+    void findById_nullId_notOk() {
+        userDao.save(defaultUser);
+        Assertions.assertThrows(DataProcessingException.class,
+                () -> userDao.findById(null));
+    }
+
+    @Test
+    void findById_incorrectId_notOk() {
+        userDao.save(defaultUser);
+        Assertions.assertThrows(NoSuchElementException.class, () -> userDao.findById(0L).get(),
+                "We can't find user by id: 0");
+        Assertions.assertThrows(NoSuchElementException.class, () -> userDao.findById(2L).get(),
+                "In DB only one record, we can't get user by id: 2");
+    }
+
+    private void setRoleDao() {
+        try {
+            userRole = new Role(Role.RoleName.USER);
+            adminRole = new Role(Role.RoleName.ADMIN);
+            roleDao.save(userRole);
+            roleDao.save(adminRole);
+        } catch (Exception e) {
+            throw new DataProcessingException("You have some problem with initialize roles, "
+                    + "fix  problem in RoleDao)", e);
+        }
```

Comment:
do you really need try catch here?

Reply:
I consider it may be usefull for users testing this code. It helps to more unsdestand our problem